rs/execution_environment/src/metrics.rs

```diff
@@ -268,8 +271,27 @@ impl QueryHandlerMetrics {
                 "The total number of transient errors accumulated \
                         during the query execution",
             ),
+            subnet_query_messages: metrics_registry.histogram_vec(
+                "execution_subnet_query_message_duration_seconds",
+                "Duration of a subnet query message execution, in seconds.",
+                decimal_buckets(-3, 2),
+                &["method_name"],
+            ),
         }
     }
+
+    pub fn observe_subnet_query_message(&self, method: &str, duration: f64) {
```

Comment:
I think this method could actually accept `QueryMethod` instead of a string (and convert to a string internally) and then we don't need any checks. These should be the only methods that can be observed as subnet queries.